public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -540,14 +540,14 @@ function addLabelFormat(
   return newQuery;
 }
 
-export function addLineFilter(query: string, value = ''): string {
+export function addLineFilter(query: string, value = '', operator = '|='): string {
   const streamSelectorPositions = getStreamSelectorPositions(query);
   if (!streamSelectorPositions.length) {
     return query;
   }
   const streamSelectorEnd = streamSelectorPositions[0].to;
 
-  const newQueryExpr = query.slice(0, streamSelectorEnd) + ` |= \`${value}\`` + query.slice(streamSelectorEnd);
+  const newQueryExpr = query.slice(0, streamSelectorEnd) + ` ${operator} \`${value}\`` + query.slice(streamSelectorEnd);
```

Comment:
https://github.com/grafana/grafana/pull/75306#discussion_r1388987019 could also be fixed in this PR.

Reply:
@svennergr do we have an "official" way to escape terms in Loki? Because this also happens in the query builder:

![imagen](https://github.com/grafana/grafana/assets/1069378/0b85f194-fe2f-4b2d-8a06-4deceacaf708)

Reply:
This wouldn't work as well: 
![imagen](https://github.com/grafana/grafana/assets/1069378/5cc638ab-026b-4ab9-98e4-fb14c06e6df8)

Reply:
Created https://github.com/grafana/grafana/issues/78100 for this.